app.py

```diff
@@ -58,13 +59,21 @@ async def get_access_token():
     token_expiry_time = datetime.now() + timedelta(seconds=expires_in)
     return access_token, token_expiry_time
 
-async def refresh_token_if_expired():
+async def refresh_access_token():
     global port_access_token, token_expiry_time, port_headers
+    logger.info("Refreshing access token...")
+    port_access_token, token_expiry_time = await get_access_token()
+    port_headers = {"Authorization": f"Bearer {port_access_token}"}
+    logger.info(f"New token received. Expiry time: {token_expiry_time}")
```

Comment:
Is there a reason you chose to reset the global variables instead of actually returning ?

Reply:
It will make it accessible for subsequent for other api calls